Terminal.Gui/ConsoleDrivers/CursesDriver/CursesDriver.cs

```diff
@@ -809,6 +474,52 @@ void ProcessInput ()
 			//}
 		}
 
+		void GetEscSeq (ref int code, ref Key k, ref int wch2, ref KeyEvent key, ref ConsoleKeyInfo [] cki)
+		{
+			ConsoleKey ck = 0;
+			ConsoleModifiers mod = 0;
+			while (code == 0) {
+				code = Curses.get_wch (out wch2);
+				var consoleKeyInfo = new ConsoleKeyInfo ((char)wch2, 0, false, false, false);
+				if (wch2 == 0 || wch2 == 27 || wch2 == Curses.KeyMouse) {
+					EscSeqUtils.DecodeEscSeq (null, ref consoleKeyInfo, ref ck, cki, ref mod, out _, out _, out _, out _, out bool isKeyMouse, out List<MouseFlags> mouseFlags, out Point pos, out _, ProcessContinuousButtonPressed);
+					if (isKeyMouse) {
+						foreach (var mf in mouseFlags) {
+							ProcessMouseEvent (mf, pos);
+						}
+						cki = null;
+						if (wch2 == 27) {
+							cki = EscSeqUtils.ResizeArray (new ConsoleKeyInfo ('\u001b', 0,
+								false, false, false), cki);
+						}
+					} else {
+						k = ConsoleKeyMapping.MapConsoleKeyToKey (consoleKeyInfo.Key, out _);
+						k = ConsoleKeyMapping.MapKeyModifiers (consoleKeyInfo, k);
+						key = new KeyEvent (k, MapKeyModifiers (k));
+						keyDownHandler (key);
+						keyHandler (key);
+					}
+				} else {
+					cki = EscSeqUtils.ResizeArray (consoleKeyInfo, cki);
+				}
+			}
+		}
+
+		void ProcessMouseEvent (MouseFlags mouseFlag, Point pos)
+		{
+			var me = new MouseEvent () {
+				Flags = mouseFlag,
+				X = pos.X,
+				Y = pos.Y
+			};
+			mouseHandler (me);
+		}
+
+		void ProcessContinuousButtonPressed (MouseFlags mouseFlag, Point pos)
```

Comment:
`ProcessContinuousButtonPress` (present tense).

Reply:
I don't mind to change it to present tense, but it's really is past tense because it was already pressed and it only be processed. Also is to make sense with the `MouseFlags.Button1Pressed` which is also past tense, like on the others drivers. I'll wait for your answer.

Reply:
Interesting. I (I guess incorrectly) thought the event meant "the button is currently (continuously) being pressed". Am I wrong?

Reply:
No you aren't incorrect. The async method is forcing to fire the button pressed event as long the button released isn't trigger and thus meaning the button pressed is really pressed. I personally prefer the `Pressed` name, but if you won't I change it, so I've also to change in all drivers, because `WindowsDriver` is using the same name.

Reply:
it's fine as-is. thanks for explaining.